beacon_node/network/src/sync/block_lookups/mod.rs

```diff
@@ -217,19 +220,29 @@ impl<T: BeaconChainTypes> BlockLookups<T> {
         // Make sure this block is not already downloaded, and that neither it or its parent is
         // being searched for.
         if let Some(parent_lookup) = self.parent_lookups.iter_mut().find(|parent_req| {
-            parent_req.contains_block(&parent_root) || parent_req.is_for_block(parent_root)
+            block_root == parent_req.chain_hash()
+                || parent_req.contains_block(&parent_root)
+                || parent_req.is_for_block(parent_root)
         }) {
             parent_lookup.add_peer(peer_id);
             // we are already searching for this block, ignore it
+            debug!(self.log, "Already searching for parent block";
+                "block_root" => ?block_root, "parent_root" => ?parent_root);
             return;
         }
 
         if self
             .processing_parent_lookups
-            .values()
-            .any(|(hashes, _peers)| hashes.contains(&block_root) || hashes.contains(&parent_root))
+            .iter()
+            .any(|(chain_hash, (hashes, _peers))| {
+                chain_hash == &block_root
+                    || hashes.contains(&block_root)
+                    || hashes.contains(&parent_root)
```

Comment:
Just to confirm my understanding here:
```rust
                chain_hash == &block_root // this means its the same parent lookup
                    || hashes.contains(&block_root) // this means we either have the parent in the hashes or the parent is already known
                    || hashes.contains(&parent_root) // kind of redundant / extra defensive check because of the above
```

Reply:
> chain_hash == &block_root // this means its the same parent lookup

the chain hash is the block triggering the parent lookup, so they should be unique per lookup. I think `hashes.contains(&parent_root)` *should* always cover this anwyways.. but this is a bit defensive.. I could be convinced to drop it.